### PR TITLE
openfortivpn: new port, version 1.6.0

### PR DESCRIPTION
**Pull request by millerdev**
#### Description

New port submission.

[Trac ticket](https://trac.macports.org/ticket/56121). Resubmitting here to make it easier for maintainers.

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.11.6 15G18013
Xcode 8.2.1 8C1002

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?

- [x] checked your Portfile with `port lint`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: net/openfortivpn/Portfile [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+PortSystem          1.0
+PortGroup           github 1.0
+
+github.setup        adrienverge openfortivpn 1.6.0 v
```
Comment (pmetzger) [inline net/openfortivpn/Portfile]: 1.7.0 is out, so why not use that?

File: net/openfortivpn/Portfile [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+github.tarball_from tags
```
Comment (pmetzger) [inline net/openfortivpn/Portfile]: Why use the tag when there are official release artifacts on github?

File: net/openfortivpn/Portfile [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+categories          net
+platforms           darwin
+license             GPL-3
+
+maintainers         @millerdev
```
Comment (pmetzger) [inline net/openfortivpn/Portfile]: You need to add your email address to this, in the form of domain:user, as in

`maintainers {foo.com:bar @millerdev}`

Unless you have a good reason to do otherwise, you should also probably add `openmaintainer`.

File: net/openfortivpn/Portfile [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+                    It spawns a pppd process and operates the communication \
+                    between the gateway and this process. It is compatible \
+                    with Fortinet VPNs.
+
+depends_build       port:autoconf \
+                    port:automake \
+                    port:pkgconfig
+depends_lib         port:openssl
+
+configure.cmd       ./autogen.sh && ./configure
```
Comment (pmetzger) [inline net/openfortivpn/Portfile]: Are you sure that one of the autotools variables (like `use_autoreconf` or what have you) doesn't cover your use case?

File: net/openfortivpn/Portfile [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+configure.pkg_config    ${prefix}/bin/pkg-config
+
+checksums           size    55622 \
+                    rmd160  348ac6ee8db8e5cf1a4cc493889e3ac3e717274e \
+                    sha256  18390e7822d562c5dcc8caa1094fdfb47ceb975f46dee862470e72e428ca75b0
```
Comment (pmetzger) [inline net/openfortivpn/Portfile]: Typically, size goes last.

File: net/openfortivpn/Portfile (hunk 2 above)
Comment (l2dy) [reply net/openfortivpn/Portfile]: @millerdev `github.tarball_from` defaults to `tags`, so you should remove this line.

@pmetzger Where are the official release artifacts? I can't find any on https://github.com/adrienverge/openfortivpn/releases.

Comment (pmetzger) [reply net/openfortivpn/Portfile]: Things like https://github.com/adrienverge/openfortivpn/archive/v1.7.0.tar.gz don't count? (Maybe I'm mistaken.( In either case, the `github.tarball_from` line isn't needed.

Comment (l2dy) [reply net/openfortivpn/Portfile]: @pmetzger That is one of two URLs of tarballs generated by GitHub/Git, and the other is the default (`tags`) used by the github PortGroup. It doesn't count as release artifacts, for which we use `releases`.

https://github.com/macports/macports-ports/blob/1f5e8dc73a1c2c2b7b4f1bf6314b2d183b1472bc/_resources/port1.0/group/github-1.0.tcl#L32-L46

**Comment by pmetzger** [comment]
@millerdev ?

File: net/openfortivpn/Portfile (hunk 1 above)
Comment (millerdev) [reply net/openfortivpn/Portfile]: The build for 1.7.0 was broken on Mac OS. That has been fixed now (https://github.com/adrienverge/openfortivpn/issues/337). I'll update to 1.7.1

**Comment by millerdev** [comment]
Updated. Thanks for the feedback, @pmetzger

I hope I got the `use_autoreconf` part right. From what I could tell, that option allowed me to remove `depends_build` as well, which is nice.

**Comment by millerdev** [comment]
I don't understand the build failure. All the tests I did locally passed, including a clean install. @pmetzger can you point out what I've done wrong?

File: net/openfortivpn/Portfile [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+description         Client for PPP+SSL VPN tunnel services.
+long_description    openfortivpn is a client for PPP+SSL VPN tunnel services. \
+                    It spawns a pppd process and operates the communication \
+                    between the gateway and this process. It is compatible \
+                    with Fortinet VPNs.
+
+depends_lib         port:openssl
+
+use_autoreconf      yes
+configure.pkg_config    ${prefix}/bin/pkg-config
```
Comment (l2dy) [inline net/openfortivpn/Portfile]: If you are using `pkg-config`, you need to add `port:pkgconfig` to `depends_build` instead of this.

File: net/openfortivpn/Portfile [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+long_description    openfortivpn is a client for PPP+SSL VPN tunnel services. \
+                    It spawns a pppd process and operates the communication \
+                    between the gateway and this process. It is compatible \
+                    with Fortinet VPNs.
+
+depends_build       port:pkgconfig
+depends_lib         port:openssl
+
+use_autoreconf      yes
+configure.pkg_config    ${prefix}/bin/pkg-config
```
Comment (l2dy) [inline net/openfortivpn/Portfile]: Why did you add the `configure.pkg_config` line? I successfully built this port without it.

**Comment by pmetzger** [comment]
Howdy! Once @l2dy's comment is addressed and the commits are squashed, I can merge this.

**Comment by millerdev** [comment]
Updated, and thank you @pmetzger and @l2dy

**Comment by pmetzger** [comment]
Merged. Thank you, @millerdev!
